app/actions/sessions.js

```diff
@@ -243,13 +242,22 @@ export function newTab() {
 export function deleteTab(tabId) {
     return function (dispatch, getState) {
         /* eslint no-alert: 0 */
+        /*
+         * Throw a dialog box at the user because deleting a connection erases
+         * it from disk permanently and persistent queries may fail onwards.
+         */
         if (confirm(DELETE_TAB_MESSAGE)) {
```

Comment:
We should move this `confirm` down into the `// If the connection has been successfully saved to disk,` block. If they haven't saved the connection yet then they don't need to be warned about deleting the tab